snmplib/transports/snmpUDPIPv4BaseDomain.c

```diff
@@ -301,6 +301,17 @@ netsnmp_udpipv4base_transport_with_source(const struct netsnmp_ep *ep,
     if (NULL == bind_addr)
         return t;
 
+    /* for Linux VRF Traps we try to bind the iface if clientaddr is not set */
+    if (ep && ep->iface && t->sock) {
+        rc =  netsnmp_bindtodevice(t->sock, ep->iface);
```

Comment:
The t->sock test looks wrong to me since 0 is a valid socket descriptor. Additionally, I think if this if-test is reached that it is already guaranteed that t->sock >= 0 and hence that it is not necessary to test t->sock.

Shouldn't netsnmp_bindtodevice() only be called if local != 0, in other words, if a socket is being created for sending data? I think local == 0 means that a listening socket is being created.

Reply:
The Linux VRF API is a little weird - they overload the "bind to device" to mean "live inside the VRF that this interface is configured to be in".  So it makes sense to bind a listen socket to a VRF, even if in the wider view it doesn't make sense to bind a listening socket to a specific interface.

Reply:
right.  you can think of the VRF as a master interface (like a bridge)...which internally, I think it is.